src/datanode/src/instance.rs

```diff
@@ -214,6 +216,36 @@ pub(crate) async fn new_object_store(store_config: &ObjectStoreConfig) -> Result
     })
 }
 
+pub(crate) async fn new_oss_object_store(store_config: &ObjectStoreConfig) -> Result<ObjectStore> {
+    let (root, secret_key, key_id, bucket, endpoint) = match store_config {
+        ObjectStoreConfig::OSS {
+            bucket,
+            root,
+            access_key_id,
+            access_key_secret,
+            endpoint,
+        } => (root, access_key_secret, access_key_id, bucket, endpoint),
+        _ => unreachable!(),
+    };
```

Comment:
Maybe we can encapsulate a struct over these configs, like `struct OssConfig` and `struct S3Config`, and pass the corresponding struct to these constructor functions.

Reply:
Yep, good idea, refactor them into such as:

```rust
pub enum ObjectStoreConfig {
    File {
        data_dir: String,
    },
    S3(S3Config),
}

#[derive(Debug, Clone, Serialize, Default, Deserialize)]
#[serde(default)]
pub struct S3Config {
    pub bucket: String,
    pub root: String,
    pub access_key_id: String,
    pub secret_access_key: String,
    pub endpoint: Option<String>,
    pub region: Option<String>,
}
```

Reply:
`File` looks very outstanding, I prefer to wrap it as well 🥹

Reply:
Hi @waynexia @killme2008 , thanks for reviewing. I have updated the code to use the structs. Please have a review again when you have time.